src/algoliasearch.helper.js

```diff
@@ -1201,7 +1201,12 @@ AlgoliaSearchHelper.prototype._search = function() {
 
   this._currentNbQueries++;
 
-  this.client.search(queries, this._dispatchAlgoliaResponse.bind(this, states, queryId));
+  this.client.search(queries)
+    .then(this._dispatchAlgoliaResponse.bind(this, states, queryId))
+    .catch(this._dispatchAlgoliaError.bind(this, queryId))
+    .catch(function(err) {
+      console.error('error', err);
```

Comment:
I'm not sure logging the error is the best thing to do at this point.

Reply:
`AlgoliaSearchHelper.emit('error', err)` seems to be closer to the original version of this code

Reply:
oh didn't read enough, I think actually the one catch should be enough

Reply:
With only the first `catch`, you can get an `UnhandledPromiseRejectionWarning`.

Reply:
maybe then just also emit an error?

Reply:
You're right! I'll do it at a higher level with a `try` `catch`.